src/main/java/io/wisetime/connector/jira/database/JiraDb.java

```diff
@@ -36,31 +47,125 @@ public void asTransaction(final Runnable runnable) {
   }
 
   public boolean canUseDatabase() {
-    // TODO
-    return true;
+    return doTableAndColumnsExist(
+        ImmutableList.of(
+            Pair.of(
+                "jiraissue",
+                ImmutableSet.of("id", "issuenum", "summary", "timespent", "project")
+            ),
+            Pair.of(
+                "project",
+                ImmutableSet.of("id", "pkey")
+            ),
+            Pair.of(
+                "cwd_user",
+                ImmutableSet.of("user_name", "lower_email_address")
+            ),
+            Pair.of(
+                "worklog",
+                ImmutableSet.of("id", "issueid", "author", "timeworked", "created", "worklogbody")
+            ),
+            Pair.of(
+                "sequence_value_item",
+                ImmutableSet.of("seq_id", "seq_name")
+            )
+        )
+    );
+  }
+
+  private boolean doTableAndColumnsExist(List<Pair<String, Set<String>>> tableAndColumnPairs) {
+    return tableAndColumnPairs.stream()
+        .allMatch(tableColumnPair -> doTableAndColumnsExist(tableColumnPair.getKey(), tableColumnPair.getValue()));
+  }
+
+  private boolean doTableAndColumnsExist(String tableName, Set<String> columnNames) {
+    boolean hasTable = query.databaseInspection()
+        .selectFromMetaData(meta -> meta.getTables(null, null, null, null))
```

Comment:
Uses  JDBC's [DatabaseMetaData](https://docs.oracle.com/javase/7/docs/api/java/sql/DatabaseMetaData.html), an interface implemented by driver vendors containing comprehensive information about the database as a whole.

Reply:
Nice, a solution that should apply to databases we care about.